### ISSUE TITLE
Merge two accounts in Cognito created with same email (email and social)

### ISSUE THREAD

**Issue by niyonx** [aws-amplify/amplify-flutter#1158]
**Describe the bug**
When a user create an account by email (ex: gmail) and later sign up by social (ex: same gmail), I have two accounts created in Cognito.

**To Reproduce**
1. sign up with email
2. sign up with facebook/ google account using the same email

**Expected behavior**
Two cognito users are created with different identity id

**Screenshots**
![image](https://user-images.githubusercontent.com/23243585/144262469-682e1434-f443-41ec-8c65-681f680439df.png)

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
- [x] Android
- [x] iOS

**Comment by niyonx** [comment]
https://github.com/aws-amplify/amplify-js/issues/369 I see this issue has been raised in amplify-js, they recommend to do a pre sign up hook. Any idea how to implement similar solution to flutter?

**Comment by fjnoyp** [comment]
Hi @niyonx thanks for reaching out to us and for providing the related js-issue - that really helped the investigation here.  

Unfortunately the pre sign up hook approach uses the AWS SDK for JavaScript and there is no equivalent AWS SDK for Dart at the moment.  I'll follow up internally about that and update you again here though. 

In the js-issue are you able to follow their instructions to change the cognito user pool setup instead?

**Comment by niyonx** [comment]
No, I have tried to add a presign up trigger from https://stackoverflow.com/questions/59635482/aws-cognito-best-practice-to-handle-same-user-with-same-email-address-signing but none of them seems to work.

I need the social sign in to either fail because of the email already assigned to a user, or allow them to sign up normally as long as they have the same identity id. Please help

**Comment by fjnoyp** [comment]
Hi thanks for your patience we'll look into a bit more later today @niyonx

**Comment by kyleVoodoo** [comment]
Hi @niyonx apologies for the later response.  We were a bit busy this week with some new developments.  I’ll see what we can next week.

**Comment by niyonx** [comment]
Cool, thank you!

**Comment by fjnoyp** [comment]
Hi @niyonx could you provide more details on how you presign up trigger is not working?  Is it not running or are there errors in your app when you implemented it?

**Comment by fjnoyp** [comment]
Hi @niyonx do you need help with this issue anymore?

**Comment by offlineprogrammer** [comment]
Hi @niyonx 

Following up on this if you can get back to us please with the details requested by @fjnoyp  above

Regards
Mo
